packages/starlight/schema.ts

```diff
@@ -147,6 +147,21 @@ export function docsSchema() {
 					 * Prevents this page from being included in autogenerated sidebar groups.
 					 */
 					hidden: z.boolean().default(false),
+					/**
+					 * Adds a badge to the sidebar link.
+					 * Can be a string or an object with a variant and text.
+					 * Variants include 'blue', 'pink', 'green', 'yellow' and 'purple'.
+					 * Passing only a string defaults to the 'blue' variant
+					 */
+					badge: z
+						.union([
+							z.string(),
+							z.object({
+								variant: z.enum(['blue', 'pink', 'green', 'yellow', 'purple']).default('blue'),
+								text: z.string(),
+							}),
+						])
+						.optional(),
```

Comment:
Maybe this could be extracted to its own file in the `packages/starlight/schemas` directory, kinda like the [prev/next links schema](https://github.com/withastro/starlight/blob/da45d02378787f8a78d78e2a81c6e6e60ae038ad/packages/starlight/schemas/prevNextLink.ts) or the [toc schema](https://github.com/withastro/starlight/blob/da45d02378787f8a78d78e2a81c6e6e60ae038ad/packages/starlight/schemas/tableOfContents.ts)?

This would mean both this file and `packages/starlight/utils/user-config.ts` could reuse the same schema without having to duplicate it, e.g.:

```ts
badge: BadgeConfigSchema(),
```

_(no idea how to create a GitHub suggestion for a potential new file)_